src/main/javascript/crypto/src/eas/EASIdentifierAttestation.ts

```diff
@@ -0,0 +1,54 @@
+import {EasTicketAttestation, EasTicketCreationOptions} from "./EasTicketAttestation";
+import {Signer} from "ethers";
+import {KeyPair, KeysArray} from "../libs/KeyPair";
+
+export interface EasIdentiferParameters {
+	version: number;
+	identifierType: "email";
+	commitment: string;
+	ethereumAddress: string;
+}
+
+export class EASIdentifierAttestation extends EasTicketAttestation {
+
+	constructor(
+		signingConfig?: {
+			EASconfig: {
+				address: string // EAS resolver contract address
+				version: string
+				chainId: number
+			},
+			signer: Signer
+		},
+		issuerKey?: KeyPair|string
+	) {
+		const schema = {
+			fields: [
+				{name: "version", type: "uint8"},
+				{name: "identifierType", type: "string"},
+				{name: "commitment", type: "bytes", isCommitment: true},
+				{name: "ethereumAddress", type: "address"}
+			]
+		};
+
+		super(schema, signingConfig, undefined, issuerKey ? <KeysArray>{"": issuerKey} : undefined);
+	}
+
+	async createEasAttestation(data: EasIdentiferParameters|any, options?: EasTicketCreationOptions, commitmentType = 'mail'){
```

Comment:
@micwallace commitment must be input value, it must be received from user side and commitmentSecret must stay in requestor side.

in current case some person, who has access to attestation.id privateKey can use it to issue attestation for ANY email. so its completely unsecure. Just check ASN version of idAttestation - it receive requestAttest and parse it for commitment and recovers userAddress

Reply:
@oleggrib Yeah but surely if the user has attestation.id private key we have bigger problems! Is there any value in keeping this client-side. What if the server just generates a random secret and creates commitment for any email address? Generating secret client side doesn't seem to offer any additional protection. Server can always issue attestations to any address it wants.

Reply:
updated code looks great.

only the question in commitment+secret+userWalletAddress.

current attestation.id version generates commitment+secret in userBrowser and signs request. it solves few problems:
1. no chance to make mistake in walletAddress, because backend recovers walletAddress from signature. backend issues attestation to the wallet, what user really controls. no chance to use wrong/fake address by mistake

2. id attestation secret stores in user browser only and dont walk through the network. it means nobody can grab network data and get idAttestation

what do you think, @micwallace ?

Reply:
Hey @oleggrib 

1. We still recover signature from Eip712AttestationRequest in order to set the correct eth address in the attestation, but this is done by common API instead of the library. You can review this code here: https://github.com/AlphaWallet/common-api/blob/85f9ec78a292c9f7cfc7547acd89f37c5509ff2a/backend/src/services/idAttestationsService.ts#L50C75-L50C75

2. We always use HTTPS so it's not possible to intercept the secret.